### ISSUE TITLE
Can't load dataset items in total size > 512MB

### ISSUE THREAD

**Issue by valekjo** [apify/apify-client-js#247]
Node has 512MB limit on string size, if the response from API is larger, the client fails to parse it.

I guess the issue might also manifest in key-value-stores? Other api endpoints probably never produce this large responses.

**Comment by mnmkng** [comment]
I don't think this is a bug really. It's quite obvious that strings of unlimited sizes cannot be loaded or parsed. That's why the list functions of client have offsets and limits and the KVS functions have `stream` options.

I would vote for closing this.

**Comment by fnesveda** [comment]
Bug report originated here: https://apifier.slack.com/archives/C0L33UM7Z/p1652976377122799
